src/codegen/mod.rs

```diff
@@ -561,7 +561,13 @@ impl CodeGenerator for Type {
                     let layout = self.layout(ctx).unwrap_or_else(Layout::zero);
                     BlobTyBuilder::new(layout).build()
                 } else {
-                    inner_item.to_rust_ty(ctx)
+                    let inner_rust_ty = inner_item.to_rust_ty(ctx);
+                    if inner_rust_ty == aster::AstBuilder::new().ty().unit() {
```

Comment:
Will do.

I was also thinking that `ToRustTy::to_rust_ty` should return an `Option` or a `Result`, and then instead of ever returning `()` we would return `None`. This is a slightly more invasive change, however, and I can do it if you want, but don't feel like going out of my way at the moment :-P